R/window.R

```diff
@@ -0,0 +1,96 @@
+# Copyright 2015 Province of British Columbia
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+# Window functions.
+rolling.sum <- function(x, width) {
+  x <- ifelse(is.na(x), 0, x)
+  cumsum(x) - lag(cumsum(x), width, default = 0)
+}
+
+rolling.mean <- function(x, width = 8, thresh = 0){
+  available <- rolling.sum(!is.na(x), width = width)
+  sum <- rolling.sum(x, width = width)
+  ifelse(available >= thresh, sum/available, NA)
+}
+
+#'number within a rolling window and step size (for dates or numeric)
+#'
+#' @import tidyr
```

Comment:
I would prefer to just import specific functions, unless using a lot of functions from the package (like we do with dplyr):

```r
#' @importFrom tidyr full_seq
```